### PR TITLE
Opt-in to Dark mode user preference

### PR DESCRIPTION
**Pull request by A6GibKm**
See https://gitlab.gnome.org/GNOME/Initiatives/-/issues/32

### CODE REVIEW

File: com.github.tchx84.Flatseal.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "app-id": "com.github.tchx84.Flatseal",
     "runtime": "org.gnome.Platform",
-    "runtime-version": "41",
+    "runtime-version": "master",
```
Comment (tchx84) [inline com.github.tchx84.Flatseal.json]: For this, we will need to wait for GNOME 42...

File: src/widgets/docsViewer.js [hunk 2]
```diff
@@ -18,6 +18,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+imports.gi.versions.WebKit2 = '4.0';
```
Comment (tchx84) [inline src/widgets/docsViewer.js]: Curious, is this related?

**Review by tchx84 — CHANGES_REQUESTED** [review]

File: com.github.tchx84.Flatseal.json (hunk 1 above)
Comment (A6GibKm) [reply com.github.tchx84.Flatseal.json]: Ill update it either when libhandy 1.6 is released or when GNOME 42 is released.

File: src/widgets/docsViewer.js (hunk 2 above)
Comment (A6GibKm) [reply src/widgets/docsViewer.js]: Yes, needed for the master runtime, otherwise WebKit will default to GTK 4, ideally we just bundle libhandy 1.6 and there is no need to use the `master` runtime.

File: com.github.tchx84.Flatseal.json (hunk 1 above)
Comment (A6GibKm) [reply com.github.tchx84.Flatseal.json]: I added libhandy 1.5.90 to the manifest rather than setting it to master.

File: src/widgets/docsViewer.js (hunk 2 above)
Comment (A6GibKm) [reply src/widgets/docsViewer.js]: This is no longer needed.

**Comment by A6GibKm** [comment]
mmm CI is failing.

Btw there is a workflow for flatpaks, see https://github.com/bilelmoussaoui/ashpd/blob/master/.github/workflows/demo-ci.yml for example.

**Comment by tchx84** [comment]
> mmm CI is failing.
> 
> Btw there is a workflow for flatpaks, see https://github.com/bilelmoussaoui/ashpd/blob/master/.github/workflows/demo-ci.yml for example.

I am aware, the problem is... Flatseal's portal tests rely on DBus services (provided by the test suite itself) that won't run with Bilal's tool.

See https://github.com/bilelmoussaoui/flatpak-github-actions/issues/25 (Ill try to find time this week and see if I can make it work)

**Comment by A6GibKm** [comment]
Updated so to use the 42 runtime which bundles libhandy. Sadly it will be years before libhandy 1.6 reaches ubuntu/debian repos.

**Comment by tchx84** [comment]
@A6GibKm as a workaround, we could install libhandy 1.6 here https://github.com/tchx84/gtk-apps-testing-docker-action/tree/gjs

**Comment by A6GibKm** [comment]
Sounds good to me, but please take care of it, CI is the bane of my existence.

**Comment by tchx84** [comment]
I pushed the changes to my CI image, but... it seems I can't re-trigger the job myself, it simply stays at "queued" forever.

**Comment by tchx84** [comment]
Still nothing..

**Comment by tchx84** [comment]
@A6GibKm I created a new PR to see if I an revive this. See https://github.com/tchx84/Flatseal/pull/416
